### PR TITLE
CSP-10 Implements locations end point

### PR DESCRIPTION
**Pull request by DevenKShah**

### CODE REVIEW

File: src/SFA.DAS.Roatp.Application.UnitTests/Locations/Queries/ProviderLocations/ProviderLocationsQueryValidatorTests.cs [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+
+namespace SFA.DAS.Roatp.Application.UnitTests.Locations.Queries.ProviderLocations
+{
+    [TestFixture]
+    public class ProviderLocationsQueryValidatorTests
+    {
+        [TestCase(10000001, true)]
+        [TestCase(10000000, false)]
+        [TestCase(100000000, false)]
+        public async Task Validate_AcceptsEightDigitNumbersOnly(int ukprn, bool expectedResult)
```
Comment (MarkFCain) [inline src/SFA.DAS.Roatp.Application.UnitTests/Locations/Queries/ProviderLocations/ProviderLocationsQueryValidatorTests.cs:16]: name doesn't match test case (it's failing on an 8 digit number 10000000.  Rename appropriately

File: src/SFA.DAS.Roatp.Application/Locations/Queries/ProviderLocationsQueryValidator.cs [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+{
+    public class ProviderLocationsQueryValidator : AbstractValidator<ProviderLocationsQuery>
+    {
+        public const string InvalidUkprnErrorMessage = "Invalid ukprn";
+        public const string ProviderNotFoundErrorMessage = "No provider found with given ukprn";
+        public ProviderLocationsQueryValidator(IProviderReadRepository providerReadRepository)
+        {
+            RuleFor(x => x.Ukprn)
+                .Cascade(CascadeMode.Stop)
+                .GreaterThan(10000000).WithMessage(InvalidUkprnErrorMessage)
```
Comment (MarkFCain) [inline src/SFA.DAS.Roatp.Application/Locations/Queries/ProviderLocationsQueryValidator.cs:14]: I think this should be GreaterThan 99999999 and less than 10000000

**Review by MarkFCain — COMMENTED** [review]
Validator and tests don't look right.

File: src/SFA.DAS.Roatp.Application.UnitTests/Locations/Queries/ProviderLocations/ProviderLocationsQueryValidatorTests.cs (hunk 1 above)
Comment (DevenKShah) [reply src/SFA.DAS.Roatp.Application.UnitTests/Locations/Queries/ProviderLocations/ProviderLocationsQueryValidatorTests.cs:16]: I am testing both the conditions, passing and failing, and the name works for both.

**Review by MarkFCain — APPROVED** [review]

**Pull request merged**